### PR TITLE
Feature/chat page

### PR DESCRIPTION
**Pull request by YaelEdri**
styling of the chat page + some basic functions

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -11,6 +11,10 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "@emotion/react": "^11.11.1",
+    "@emotion/styled": "^11.11.0",
```
Comment (asafcohen98) [inline client/package.json]: @YaelEdri - Why did you install `emotion`?

File: client/src/App.tsx [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import {} from 'react';
```
Comment (asafcohen98) [inline client/src/App.tsx]: @YaelEdri - Remove this line

File: client/src/App.scss [hunk 3]
```diff
@@ -1,12 +1,23 @@
 @import './styles/variables.scss';
 
+@font-face {
+  font-family: 'Assistant';
+  src:
+    local('AssistantRegular'),
+    url('./fonts/Assistant.ttf') format('truetype');
+  font-weight: normal;
+}
+
```
Comment (asafcohen98) [inline client/src/App.scss]: @YaelEdri - I think that you need to split it to another PR: 'Set up font faces"

File: client/src/App.scss [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
   height: 100vh;
   width: 100vw;
+  direction: rtl;
+  font-size: 16px;
 
   * {
     box-sizing: border-box;
-    display: inline-block;
+    // display: inline-block;
+    font-family: 'Assistant';
```
Comment (asafcohen98) [inline client/src/App.scss]: @YaelEdri - It is not related to this task, this is a global setup.

File: client/src/components/ChatInput.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import SendRoundedIcon from '@mui/icons-material/SendRounded';
+
+export interface ChatInputProps {
+  sendMsgFunc: (text: string) => void;
+}
```
Comment (asafcohen98) [inline client/src/components/ChatInput.tsx]:
```suggestion
export interface IChatBoxProps {
  sendChatMsg: (text: string) => void;
}
```

File: client/src/components/ChatInput.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+export const ChatInput: FC<ChatInputProps> = ({ sendMsgFunc }: ChatInputProps) => {
```
Comment (asafcohen98) [inline client/src/components/ChatInput.tsx]:
```suggestion
export const ChatBox: FC<IChatBoxProps> = ({ sendChatMsg }) => {
```

File: client/src/components/ChatInput.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  const [input, setInput] = useState<string>('');
```
Comment (asafcohen98) [inline client/src/components/ChatInput.tsx]:
```suggestion
  const [currentMsg, setCurrentMsg] = useState<string>('');
```

File: client/src/components/ChatInput.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+          onChange={setInputOnChange}
+          onKeyDown={checkEnterPress}
+          placeholder="הודעה"
+        />
+        <button className="chat-input-send" onClick={() => sendMsg()}>
+          <SendRoundedIcon />
+        </button>
+      </div>
+    </div>
+  );
```
Comment (asafcohen98) [inline client/src/components/ChatInput.tsx]:
```suggestion
  return (
    <div className="chat-box_container">
      <form className="chat-box_form" onSubmit={onSendMsg}>
        <input
          className="chat-box_input"
          value={currentMsg}
          onChange={handleChange}
          placeholder="הודעה"
        />
        <button className="chat-box_submit-btn" type="submit">
          <SendRoundedIcon />
        </button>
      </form>
    </div>
  );
```

File: client/src/components/ChatInput.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    if (e.key === 13) {
+      sendMsg();
+    }
+    return;
+  };
+
+  const setInputOnChange = (e: React.FormEvent<HTMLInputElement>) => {
+    setInput(e.target.value);
+    return;
+  };
```
Comment (asafcohen98) [inline client/src/components/ChatInput.tsx]:
```suggestion
  const handleChange = (e: React.FormEvent<HTMLInputElement>) => {
    setCurrentMsg(e.target.value);
  };
```

File: client/src/components/ChatInput.tsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    //add validation of input
+    sendMsgFunc?.(input);
+  };
+
+  const checkEnterPress = (e: KeyboardEvent) => {
+    if (e.key === 13) {
+      sendMsg();
+    }
+    return;
+  };
```
Comment (asafcohen98) [inline client/src/components/ChatInput.tsx]: @YaelEdri - You need to wrap the input and the button with the form and remove this logic.

File: client/src/components/ChatInput.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  sendMsgFunc: (text: string) => void;
+}
+
+export const ChatInput: FC<ChatInputProps> = ({ sendMsgFunc }: ChatInputProps) => {
+  const [input, setInput] = useState<string>('');
+
+  const sendMsg = () => {
+    //add validation of input
+    sendMsgFunc?.(input);
+  };
```
Comment (asafcohen98) [inline client/src/components/ChatInput.tsx]:
```suggestion
  const onSendMsg = () => {
    //add validation of input
    sendChatMsg?.(currentMsg);
  };
```

File: client/src/components/ChatTopBar.tsx [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+  endChat?: () => void;
+  replace?: () => void;
+}
+
+export const ChatTopBar: FC<ChatTopBarProps> = ({
+  chatEnded,
+  isSupporter,
+  endChat,
+  replace
+}: ChatTopBarProps) => {
```
Comment (asafcohen98) [inline client/src/components/ChatTopBar.tsx]:
```suggestion
interface IChatTopBarProps {
  isChatEnded: boolean;
  isSupporter: boolean;
  endChat?: () => void;
  changeChatRoom?: () => void;
}

export const ChatTopBar: FC<IChatTopBarProps> = ({
  isChatEnded,
  isSupporter,
  endChat,
  changeChatRoom
}: IChatTopBarProps) => {
```

File: client/src/components/ChatTopBar.tsx [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+              }}
+            >
+              סיום שיחה
+            </button>
+          </div>
+        )}
+      </div>
+    </div>
+  );
+};
```
Comment (asafcohen98) [inline client/src/components/ChatTopBar.tsx]: @YaelEdri - The same as the other comments, I think we need to agree about CSS class name conventions.

File: client/src/components/Message.tsx [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+export interface MessageProps {
+  id: number;
```
Comment (asafcohen98) [inline client/src/components/Message.tsx]: @YaelEdri - The id prop is not relevant you can pass it anyway because it's part of react component props. just remove it.

File: client/src/pages/Chat.tsx [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+import { ChatTopBar } from '../components/ChatTopBar';
+import { ChatInput } from '../components/ChatInput';
+import '../styles/Chat.scss';
+
+export const Chat = () => {
+  const [msg, setMsg] = useState<MessageProps[]>([
+    { isSender: false, content: 'הודעה ראשונה', id: 1 },
+    { isSender: false, content: 'הודעה שנייה', id: 2 },
+    { isSender: true, content: 'תגובה ראשונה אמור להיות בסדר ', id: 3 }
+  ]);
```
Comment (asafcohen98) [inline client/src/pages/Chat.tsx]: @YaelEdri - You can create a mock file, with this data, it will be cleaner.

File: client/src/pages/Chat.tsx [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+
+  const msgIsEmpty = useMemo(() => msg.length == 0, msg);
```
Comment (asafcohen98) [inline client/src/pages/Chat.tsx]: @YaelEdri - `useMemo`  is for really big calculations, in that case, you can just create a regular const.

File: client/src/styles/Chat.scss
Comment (asafcohen98) [inline client/src/styles/Chat.scss:1]: @YaelEdri - We have to talk about CSS conventions, to declare `spacing` variables, and to split scss file for every component, it could be better.

**Review by asafcohen98 — DISMISSED** [review]
See comments and suggestions

File: client/src/App.scss (hunk 3 above)
Comment (YaelEdri) [reply client/src/App.scss]: No problem

File: client/src/App.scss (hunk 4 above)
Comment (YaelEdri) [reply client/src/App.scss]: This is a setup that I added in one of my prev branches that I remembered from a previous project I worked on that may be useful but, eventually it only got in the way so I deleted it in this branch...
If you want me to bring it back let me know

File: client/src/components/ChatInput.tsx (hunk 6 above)
Comment (YaelEdri) [reply client/src/components/ChatInput.tsx:9]: May I know the reason to the change in the name?

File: client/package.json (hunk 1 above)
Comment (YaelEdri) [reply client/package.json]: I uninstalled them.

File: client/src/components/ChatInput.tsx (hunk 5 above)
Comment (YaelEdri) [reply client/src/components/ChatInput.tsx]: Changed it locally but removed the I.

File: client/src/components/ChatInput.tsx (hunk 6 above)
Comment (YaelEdri) [reply client/src/components/ChatInput.tsx]: Changed it locally but removed the I.

File: client/src/components/ChatTopBar.tsx (hunk 13 above)
Comment (YaelEdri) [reply client/src/components/ChatTopBar.tsx]: Once there will be a convention I will happily change it

File: client/src/components/Message.tsx (hunk 14 above)
Comment (YaelEdri) [reply client/src/components/Message.tsx]: Yes, i forgot to delete it when I finished with it...

File: client/src/pages/Chat.tsx (hunk 15 above)
Comment (YaelEdri) [reply client/src/pages/Chat.tsx]: OK, I will also create a folder for it

File: client/src/styles/Chat.scss
Comment (YaelEdri) [reply client/src/styles/Chat.scss:1]: No problem. I did what I could do to the lack of conventions but when there will be ones I will gladly change it.
But I'm not quite sue what you meant when you wrote "spacing". do you mean the margin' padding and alike?
Also' I changed the styling a bit to be more responsive.

**Review by TheLastVigil — APPROVED** [review]
in order to move on with other parts of the application, I think we should approve and merge.


it seems like all the issues were resolved?

**Review by RealByter — APPROVED** [review]
